### PR TITLE
Caching out-of-cluster cost requests

### PR DESCRIPTION
**Pull request by nikovacevic**
Branched off of [PR 204](https://github.com/kubecost/cost-model/pull/204), so that is a prerequisite to this for all intents and purposes.

### CODE REVIEW

**Comment by dwbrown2** [comment]
Just wanted to clarify... are you removing the ability to pass start and end date to the aggregatedCostModel? That's fine if so, just wanted to make sure I understand.

**Comment by nikovacevic** [comment]
> Just wanted to clarify... are you removing the ability to pass start and end date to the aggregatedCostModel? That's fine if so, just wanted to make sure I understand.

@dwbrown2 Correct. At least until we come up with a sophisticated cache that can pull chunks out of a cached timeline, this makes the logic of when to cache and how more simpler and thus better IMHO.

Any objections? (e.g. does this screw up the frontend's custom date range querying?) It wouldn't be hard to put it back and set a rule to avoid the cache if the time range is defined by start/end rather than window/offset.

**Comment by dwbrown2** [comment]
Nope, this shouldn't be a problem. Just wanted to know so I can communicate
correctly to new users.

Webb
Kubecost, Co-founder

ᐧ

File: costmodel/cluster.go [hunk 1]
```diff
@@ -142,7 +142,7 @@ func resultToTotal(qr interface{}) (map[string][][]string, error) {
 }
 
 // ClusterCostsForAllClusters gives the cluster costs averaged over a window of time for all clusters.
-func ClusterCostsForAllClusters(cli prometheusClient.Client, cloud costAnalyzerCloud.Provider, windowString, offset string) (map[string]*Totals, error) {
+func ClusterCostsForAllClusters(cli prometheusClient.Client, windowString, offset string) (map[string]*Totals, error) {
```
Comment (AjayTripathy) [inline costmodel/cluster.go]: We actually do need this, see: https://github.com/kubecost/cost-model/pull/210/files#diff-58d9ebd2d60cdef59d19220180531c6fR151

File: costmodel/router.go [hunk 2]
```diff
@@ -260,169 +326,100 @@ func (a *Accesses) AggregateCostModel(w http.ResponseWriter, r *http.Request, ps
...
 
+	// enable remote if it is available and not disabled
+	remoteAvailable := os.Getenv(remoteEnabled) == "true"
+	remoteEnabled := remote && remoteAvailable
+	klog.Infof("REMOTE ENABLED: %t", remoteEnabled)
+
+	// default to given Prometheus client, but use Thanos Client if it
+	// exists and remote is not disabled
+	pClient := a.PrometheusClient
+	if remoteEnabled && a.ThanosClient != nil {
```
Comment (AjayTripathy) [inline costmodel/router.go]: just use a.ThanosClient != nil here. The remoteEnabled flag is for postgres. ThanosClient will only be instantiated if Thanos is available.

Comment (nikovacevic) [reply costmodel/router.go]: That would be a change from master (see https://github.com/kubecost/cost-model/blob/master/costmodel/router.go#L414). Is that still the intended logic?

File: costmodel/cluster.go (hunk 1 above)
Comment (nikovacevic) [reply costmodel/cluster.go]: Ok I'll put that back

**Pull request closed without merging**